salt/transport/zeromq.py

```diff
@@ -810,6 +810,34 @@ def publish(self, load):
 
 
 # TODO: unit tests!
+class AsyncReqMessageClientPool(object):
+    def __init__(self, opts, addr, linger=0, io_loop=None, socket_pool=20):
```

Comment:
As I think the pool size should be configurable. And the default value should be either `1` to keep the current behavior untouched, because this is unneeded for small salt configurations.